cmd/oras/manifest/fetchConfig.go

```diff
@@ -0,0 +1,172 @@
+Example - Fetch the descriptor of the config:
+oras manifest fetch-config localhost:5000/hello:latest --descriptor
+`,
+		Args: cobra.ExactArgs(1),
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			opts.cacheRoot = os.Getenv("ORAS_CACHE")
+			return opts.ReadPassword()
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.targetRef = args[0]
+			return fetchConfig(opts)
+		},
+	}
+
+	cmd.Flags().StringVarP(&opts.outputPath, "output", "o", "", "output file path")
+	cmd.Flags().StringVarP(&opts.mediaType, "media-type", "", "", "media type of the manifest config")
+	option.ApplyFlags(&opts, cmd.Flags())
+	return cmd
+}
+
+func fetchConfig(opts fetchConfigOptions) (err error) {
+	ctx, _ := opts.SetLoggerLevel()
+
+	repo, err := opts.NewRepository(opts.targetRef, opts.Common)
+	if err != nil {
+		return err
+	}
+
+	if repo.Reference.Reference == "" {
+		return errors.NewErrInvalidReference(repo.Reference)
+	}
+
+	var src oras.ReadOnlyTarget = repo
+	if opts.cacheRoot != "" {
+		ociStore, err := oci.New(opts.cacheRoot)
+		if err != nil {
+			return err
+		}
+		src = cache.New(repo, ociStore)
+	}
+
+	configDesc, err := fetchConfigDesc(ctx, src, opts.targetRef, opts.mediaType)
+	if err != nil {
+		return err
+	}
+
+	// outputs config's descriptor if `--descriptor` is used
+	if opts.OutputDescriptor {
+		descBytes, err := json.Marshal(configDesc)
+		if err != nil {
+			return err
+		}
+		err = opts.Output(os.Stdout, descBytes)
+		if err != nil {
+			return err
+		}
+	}
+
+	contentBytes, err := content.FetchAll(ctx, src, configDesc)
+	if err != nil {
+		return err
+	}
+
+	// save config into the local file if the output path is provided
+	if opts.outputPath != "" {
+		file, err := os.OpenFile(opts.outputPath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, os.ModePerm)
+		if err != nil {
+			return err
+		}
+		defer func() {
+			if closeErr := file.Close(); err == nil {
+				err = closeErr
+			}
+		}()
+
+		err = opts.Output(file, contentBytes)
+		if err != nil {
+			return err
+		}
+	}
+
+	if !opts.OutputDescriptor && opts.outputPath == "" {
+		err = opts.Output(os.Stdout, contentBytes)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func fetchConfigDesc(ctx context.Context, src oras.ReadOnlyTarget, reference string, configMediaType string) (ocispec.Descriptor, error) {
+	// fetch manifest descriptor
+	manifestDesc, err := oras.Resolve(ctx, src, reference, oras.ResolveOptions{})
+	if err != nil {
+		return ocispec.Descriptor{}, err
+	}
+
+	// fetch config descriptor
+	successors, err := content.Successors(ctx, src, manifestDesc)
+	if err != nil {
+		return ocispec.Descriptor{}, err
+	}
+	for i, s := range successors {
+		if s.MediaType == configMediaType || (configMediaType == "" && i == 0 && isImageManifest(manifestDesc.MediaType)) {
+			return s, nil
+		}
+	}
+	return ocispec.Descriptor{}, fmt.Errorf("%s does not have a config", reference)
+}
+
+func isImageManifest(mediaType string) bool {
```

Comment:
Consider moving this function to utils file or under internal folder to avoid duplicate code.
https://github.com/oras-project/oras/blob/main/cmd/oras/pull.go#L185